planner/core/rule_join_elimination.go

```diff
@@ -69,59 +85,68 @@ func (o *outerJoinEliminator) extractInnerJoinKeys(join *LogicalJoin, innerChild
 	return expression.NewSchema(joinKeys...)
 }
 
-func (o *outerJoinEliminator) isAggColsAllFromOuterTable(outerPlan LogicalPlan, aggCols []*expression.Column) bool {
+func (o *outerJoinEliminator) isAggColsAllFromOuterTable(outerPlan LogicalPlan, aggCols []*expression.Column) (bool, error) {
 	if len(aggCols) == 0 {
-		return false
+		return false, nil
 	}
 	for _, col := range aggCols {
 		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
-		if c, err := outerPlan.Schema().FindColumn(columnName); c == nil {
-			log.Warn(err)
-			return false
+		c, err := outerPlan.Schema().FindColumn(columnName)
+		if err != nil {
```

Comment:
Can we merge this `if` with the following `if` by:
```go
if err != nil || c == nil {
	return false, err
}
```

Reply:
Yes, it is better.